nni/contrib/compression/quantization/lsq_quantizer.py

```diff
@@ -0,0 +1,124 @@
+    evaluator
+        TODO: {evaluator_docstring}
+    quant_start_step
+        The steps for warmup training before QAT begin.
+
+    Examples
+    --------
+        >>> from nni.contrib.compression.quantization import LsqQuantizer
+        >>> from nni.contrib.compression.utils import TorchEvaluator
+        >>> model = ...
+        >>> optimizer = ...
+        >>> max_steps, max_epochs = ..., ...
+        >>> evaluator = TorchEvaluator(train, optimizer, training_step)
+        >>> quantizer = LsqQuantizer(model, configure_list, evaluator)
+        >>> _, calibration_config = quantizer.compress(max_steps, max_epochs)
+    '''
+    def __init__(self, model: torch.nn.Module, config_list: List[Dict], evaluator: Evaluator, \
+                 existed_wrappers: Dict[str, ModuleWrapper] | None = None):
+        super().__init__(model, config_list, evaluator, existed_wrappers=existed_wrappers)
+        self.evaluator: Evaluator
+        self.is_init = False
+
+        self.register_scale()
+        self.register_lsq_apply_method()
+        self.register_track_func()
+
+    def register_track_func(self):
+        for module_name, _ in self._target_spaces.items():
+            wrapper = self._module_wrappers[module_name]
+            wrapper.register_track_func(self.init_scale)
+
+    def init_scale(self, wrapper: ModuleWrapper, target_name: str, target: Tensor):
+        if self.is_init or target_name not in wrapper.quantization_target_spaces:
+            return
+        target_space = wrapper.quantization_target_spaces[target_name]
+        init_target = target.data.detach().abs().mean() * 2 / (target_space.qmax ** 0.5)
+        target_space.scale.data = init_target # type: ignore
+        target_space.zero_point = torch.tensor(0.0).to(target.device)
+
+    def register_lsq_apply_method(self):
+        for _, ts in self._target_spaces.items():
+            for _, target_space in ts.items():
+                target_space.apply_method = "lsq_clamp_round"
+
+    def register_scale(self):
+        for module_name, ts in self._target_spaces.items():
+            wrapper = self._module_wrappers[module_name]
+            for target_name, _ in ts.items():
+                if hasattr(wrapper, f"{target_name}_scale"):
+                    delattr(wrapper, f"{target_name}_scale")
+                try:
+                    device = next(wrapper.parameters()).device
+                except StopIteration:
+                    try:
+                        device = next(wrapper.buffers()).device
+                    except StopIteration:
+                        # NOTE: this will have risk in model parallel
+                        device = next(self.bound_model.parameters()).device
+                param = torch.nn.Parameter(torch.Tensor([1.0]).to(device))
+                wrapper.register_parameter(f"{target_name}_scale", param)
+
+    def patch_optimizer_param_group(self):
```

Comment:
There are two possible suggestion:
1. call `super().patch_optimizer_param_group` to get the bias param group, then merge with scale param group, or put them to a param group list.
2. rename this func and call these param group creation functions one by one.